File: app/src/main/java/com/example/yue/nexttext/Data/Message.java
package com.example.yue.nexttext.Data;

import android.os.Parcel;
import android.os.Parcelable;


/**
 * Created by jamesmulvenna on 2017-09-28.
 */

public class Message implements Parcelable {
    private String from, password, to, subject, message;

    public Message() {}

    // For email
    public Message(String newFrom, String newPassword, String newTo, String newSubject, String newMessage){
        this.from = newFrom;
        this.password = newPassword;
        this.to = newTo;
        this.subject = newSubject;
        this.message = newMessage;
    }

    // For sms
    public Message(String newTo, String newMessage){
        this.to = newTo;
        this.message = newMessage;
    }

    protected Message(Parcel in) {
        from = in.readString();
        to = in.readString();
        subject = in.readString();
        message = in.readString();
    }

    public static final Creator<Message> CREATOR = new Creator<Message>() {
        @Override
        public Message createFromParcel(Parcel in) {
            return new Message(in);
        }

        @Override
        public Message[] newArray(int size) {
            return new Message[size];
        }
    };

    public String getFrom(){ return this.from; }
    public void setFrom(String newFrom) { this.from = newFrom; }
    public String getPassword(){ return this.password; }
    public void setPassword(String newPassword) { this.password = newPassword; }
    public String getTo(){ return this.to; }
    public void setTo(String newTo) { this.to = newTo; }
    public String getSubject(){ return this.subject; }
    public void setSubject(String newSubject) { this.subject = newSubject; }
    public String getMessage(){ return this.message; }
    public void setMessage(String newMessage) { this.message = newMessage; }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(to);
        parcel.writeString(subject);
        parcel.writeString(message);
    }
}

File: app/src/main/java/com/example/yue/nexttext/Data/Message.kt
package com.example.yue.nexttext.Data

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by yue on 2017-10-26.
 * A class can be used to hold either for SMS or Email data
 *
 */
class Message() : Parcelable {
    var from: String? = null
    var password: String? = null
    var to: String? = null
    var subject: String? = null
    var message: String? = null

    constructor(parcel: Parcel) : this() {
        from = parcel.readString()
        password = parcel.readString()
        to = parcel.readString()
        subject = parcel.readString()
        message = parcel.readString()
    }

    // For email
    // TODO: Actual constructor shouldn't take null value, they are here just because the whole project isn't complete
    constructor(_from: String?, _password: String?, newTo: String?, _subject: String?, _message: String?) : this() {
        this.from = _from
        this.password = _password
        this.to = newTo
        this.subject = _subject
        this.message = _message
    }

    constructor(_to: String, _subject: String, _message: String) : this() {
        this.to = _to
        this.subject = _subject
        this.message = _message
    }

    // For sms
    constructor(_to: String, _message: String) : this() {
        this.to = _to
        this.message = _message
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(from)
        parcel.writeString(password)
        parcel.writeString(to)
        parcel.writeString(subject)
        parcel.writeString(message)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<Message> {

        override fun createFromParcel(parcel: Parcel): Message = Message(parcel)

        override fun newArray(size: Int): Array<Message?> = arrayOfNulls(size)
    }
}